syncer/checkpoint.go

```diff
@@ -139,14 +139,14 @@ func (b *binlogPoint) outOfDate() bool {
 func (b *binlogPoint) MySQLLocation() binlog.Location {
 	b.RLock()
 	defer b.RUnlock()
-	return b.location.Clone()
+	return b.location
```

Comment:
How about adding some comments to tell the caller to clone it for writer operations? if so, ditto for `FlushedMySQLLocation`. may also for other public methods?

Reply:
Use SetGTID rather than GITD.Set always. We may change attributes in Location to lowercase later and don't need to clone.